mrjob/dataproc.py

```diff
@@ -0,0 +1,873 @@
+
+    Input, support, and jar files can be either local or on GCS; use
+    ``gs://...`` URLs to refer to files on GCS.
+
+    This class has some useful utilities for talking directly to GCS and Dataproc,
+    so you may find it useful to instantiate it without a script::
+
+        from mrjob.dataproc import DataprocJobRunner
+        ...
+    """
+    alias = 'dataproc'
+
+    # Don't need to bootstrap mrjob in the setup wrapper; that's what
+    # the bootstrap script is for!
+    BOOTSTRAP_MRJOB_IN_SETUP = False
+
+    OPTION_STORE_CLASS = DataprocRunnerOptionStore
+
+    def __init__(self, **kwargs):
+        """:py:class:`~mrjob.dataproc.DataprocJobRunner` takes the same arguments as
+        :py:class:`~mrjob.runner.MRJobRunner`, plus some additional options
+        which can be defaulted in :ref:`mrjob.conf <mrjob.conf>`.
+\
+        """
+        super(DataprocJobRunner, self).__init__(**kwargs)
+
+        # if we're going to create a bucket to use as temp space, we don't
+        # want to actually create it until we run the job (Issue #50).
+        # This variable helps us create the bucket as needed
+
+        # TODO - mtai @ davidmarin - We should re-instance the bucket creation logic used in EMR, instead creating buckets in GCS
+        self._fix_gcs_tmp_opts()
+
+        # use job key to make a unique tmp dir
+        self._gcs_tmp_dir = self._opts['gcs_tmp_dir'] + self._job_key + '/'
+
+        # pick/validate output dir
+        if self._output_dir:
+            self._output_dir = self._check_and_fix_gcs_dir(self._output_dir)
+        else:
+            self._output_dir = self._gcs_tmp_dir + 'output/'
+
+        # manage working dir for bootstrap script
+        self._bootstrap_dir_mgr = BootstrapWorkingDirManager()
+
+        # manage local files that we want to upload to GCS. We'll add them
+        # to this manager just before we need them.
+        gcs_files_dir = self._gcs_tmp_dir + 'files/'
+        self._upload_mgr = UploadDirManager(gcs_files_dir)
+
+        self._bootstrap = self._bootstrap_python() + self._parse_bootstrap()
+
+        for cmd in self._bootstrap:
+            for maybe_path_dict in cmd:
+                if isinstance(maybe_path_dict, dict):
+                    self._bootstrap_dir_mgr.add(**maybe_path_dict)
+
+        # we'll create the script later
+        self._master_bootstrap_script_path = None
+
+        self._gcp_project = self._opts['gcp_project']
+        self._gcp_region = self._opts['gcp_region']
+        self._gcp_zone = self._opts['gcp_zone']
+        self._gcp_cluster = self._opts['gcp_cluster']
+
+        self._dataproc_client = None
+
+        # when did our particular task start?
+        self._dataproc_job_start = None
+
+        # This will be filled by _wait_for_steps_to_complete()
+        self._log_interpretations = []
+
+    @property
+    def dataproc_client(self):
+        if not self._dataproc_client:
+            credentials = GoogleCredentials.get_application_default()
```

Comment:
Ah, this is how the credentials are added.

You probably at least want a link to https://developers.google.com/identity/protocols/application-default-credentials#howtheywork. And it should be possible to specify the credentials in `mrjob.conf`.